File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/arithmetic/ZeroInstruction.java
package com.cristiano.alife.instructions.arithmetic;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

public class ZeroInstruction extends BaseInstruction {
	public ZeroInstruction(IWorld mundo) {
		super(mundo);
	}

	@Override
	public void executa(IOrganismo o) {
		o.setReg(getByteOrganismo(o, 1), 0);
	}

	@Override
	public int getStep() {
		return 2;
	}

	@Override
	public String getDescription(IOrganismo o, int ip) {
		return comment(name+" "+ALifeConsts.getLetter(o.getMemory(ip+1)), ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <- 0");
	}

}

File: netAvida/netAvida.Backend/instructions/impl/arithmetic/ZeroInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class ZeroInstruction : BaseInstruction
    {
        public ZeroInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            o.setReg(getByteOrganismo(o, 1), 0);
        }

    public override int getStep()
        {
            return 2;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)), ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <- 0");
        }
    }
}
